File: superstitioMain/java/superstitio/delayHpLose/DelayHpLosePower_ApplyOnlyOnVictory.java
package superstitio.delayHpLose;

import com.megacrit.cardcrawl.core.AbstractCreature;
import superstitio.DataManager;

public class DelayHpLosePower_ApplyOnlyOnVictory extends DelayHpLosePower {
    public static final String POWER_ID = DataManager.MakeTextID(DelayHpLosePower_ApplyOnlyOnVictory.class);

    public DelayHpLosePower_ApplyOnlyOnVictory(final AbstractCreature owner, int amount) {
        super(POWER_ID, owner, amount);
    }

    @Override
    public boolean checkShouldInvisibleTips() {
        return false;
    }

    @Override
    public void onVictory() {
        immediate_applyDamage(this);
    }

    @Override
    protected int addToBot_removeDelayHpLoss(int amount, boolean removeOther) {
        int lastAmount = amount - this.amount;
        addToBot_reducePowerToOwner(this.ID, amount);
        return lastAmount;
    }

    @Override
    public boolean showDecreaseAmount() {
        return true;
    }
}

File: superstitioMain/java/superstitio/delayHpLose/DelayHpLosePower_ApplyOnlyOnVictory.kt
package superstitio.delayHpLose

import com.megacrit.cardcrawl.core.AbstractCreature
import superstitio.DataManager
import superstitioapi.utils.addToBot_reducePowerToOwner

class DelayHpLosePower_ApplyOnlyOnVictory(owner: AbstractCreature, amount: Int) :
    DelayHpLosePower(POWER_ID, owner, amount)
{
    override fun checkShouldInvisibleTips(): Boolean
    {
        return false
    }

    override fun onVictory()
    {
        immediate_applyDamage(this)
    }

    override fun showDecreaseAmount(): Boolean
    {
        return true
    }

    override fun addToBot_removeDelayHpLoss(amount: Int, removeOther: Boolean): Int
    {
        val lastAmount = amount - this.amount
        addToBot_reducePowerToOwner(this.ID, amount)
        return lastAmount
    }

    companion object
    {
        val POWER_ID: String = DataManager.MakeTextID(DelayHpLosePower_ApplyOnlyOnVictory::class.java)
    }
}
